### PR TITLE
feat: implements kubernetes upgrade/downgrade functionality

### PR DESCRIPTION
**Pull request by baranbartu**
Implements `Kubernetes` upgrade/downgrade functionality.

```
(syntho-cli-py3.12) (⎈|kind-syntho-test-3:syntho)➜  cli git:(feat/kubernetes-upgrade-downgrade) ✗ syntho-cli k8s update --deployment-id k8s-77eaac704cb8c4ebf90c0a7ccf2be57c --new-version 1.28.0
-- Syntho stack is going to be updated from (1.29.0) to (1.28.0) (Kubernetes) --

Step 1: Further compatibility analysis is being made;
        - [done] Fetching the releases: 1.29.0 vs 1.28.0
        - [done] Checking if the releases are compatible: 1.29.0 vs 1.28.0
        - [done] Cleanup tempoary files and directories
Step 2: Rolling out new release;
        - [done] Fetching desired version if it was not previously rolled out: 1.28.0
        - [done] Rolling out release from 1.29.0 to 1.28.0

The application stack has been successfully rolled out to a given version. See helpful commands below.

Deployment status: syntho-cli k8s status --deployment-id k8s-77eaac704cb8c4ebf90c0a7ccf2be57c
Destroy deployment: syntho-cli k8s destroy --deployment-id k8s-77eaac704cb8c4ebf90c0a7ccf2be57c
Update release: syntho-cli k8s update --deployment-id k8s-77eaac704cb8c4ebf90c0a7ccf2be57c --new-version <version>
See all releases: syntho-cli releases

```

### CODE REVIEW

File: cli/cli/scripts/update-release.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+do_rollout_docker_compose() {
+    echo "TBI"
+}
+
+rollout_failure_callback() {
+    with_loading "Deployment rollout to new release has been timedout." do_nothing "" "" 2
+    with_loading "However, please check pods in syntho namespace, perhaps the deployment is still being rolled out." do_nothing "" "" 2
+    with_loading "Contact support@syntho.ai in case the issue persists." do_nothing "" "" 2
```
Comment (Younday) [inline cli/cli/scripts/update-release.sh:103]: Could we put the support email address somewhere as a global variable/config? Just to make it easier to change

**Review by Younday — COMMENTED** [review]
Looks good in general, just a few small things. Are there also particular tests that we can add for this change or is that harder to do in this case? I see only test has been changed to have the updated response from the CLI

File: cli/cli/scripts/compatibility-check.sh [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        errors+="Unexpected failure when cleaning up stale files and directories\n"
+    fi
+
+    write_and_exit "$errors" "cleanup_files_and_directories"
+}
+
+
+with_loading "Fetching the releases: $CURRENT_VERSION vs $NEW_VERSION" fetch_releases
+with_loading "Checking if the releases are compatible: $CURRENT_VERSION vs $NEW_VERSION" check_if_compatible
+with_loading "Cleanup tempoary files and directories" cleanup_files_and_directories
```
[the remark this answers is not in the document]
Comment (baranbartu) [reply cli/cli/scripts/compatibility-check.sh]: done @Younday

File: cli/cli/scripts/update-release.sh (hunk 1 above)
Comment (baranbartu) [reply cli/cli/scripts/update-release.sh]: done @Younday

**Comment by baranbartu** [comment]
> Looks good in general, just a few small things. Are there also particular tests that we can add for this change or is that harder to do in this case? I see only test has been changed to have the updated response from the CLI

I think tests always can be added. In this case, My inclination is to add tests for syntho-cli user interaction part first. Then later, we can add an integration test for this one. WDYT?

**Comment by Younday** [comment]
Yeah that's fine with me, the integration test for this one also probably will get a bit more complicated.

**Review by Younday — APPROVED** [review]

**Comment by baranbartu** [comment]
> Yeah that's fine with me, the integration test for this one also probably will get a bit more complicated.

added some tests

**Pull request merged**